Rubberduck.Core/AutoComplete/AutoCompleteFunctionBlock.cs

```diff
@@ -22,11 +22,23 @@ public override bool Execute(AutoCompleteEventArgs e, AutoCompleteSettings setti
                 using (var pane = module.CodePane)
                 {
                     var original = module.GetLines(e.CurrentSelection);
-                    var asTypeClause = $" {Tokens.As} {Tokens.Variant}";
-                    var code = original + (Regex.IsMatch(original, $"\\) {Tokens.As} ") ? string.Empty : asTypeClause);
+                    var hasAsToken = Regex.IsMatch(original, $"\\)\\s+{Tokens.As}", RegexOptions.IgnoreCase) ||
+                                     Regex.IsMatch(original, $"{Tokens.Function}\\s+\\(.*\\)\\s+{Tokens.As} ", RegexOptions.IgnoreCase);
+                    var hasAsType = Regex.IsMatch(original, $"{Tokens.Function}\\s+\\w+\\(.*\\)\\s+{Tokens.As}\\s+\\w+", RegexOptions.IgnoreCase);
```

Comment:
All those regex expressions give me headache. It might be better to encapsulate each regex expression in a function for readable code? Also because I'm not sure about some expression, particularly the `\\s+{Tokens.As}"` on the first line -- wouldn't we want to check for a space after the `As`, too?

Reply:
BTW, this might be a case for using the string literal (`@$"..."`) to avoid having to double-escape the regex expressions.

Reply:
@bclothier I can't think of a legal bit of VBA code that matches a closing parenthesis followed by a space and an `As` token, *on a line of code that can safely be assumed to be a member signature*, that isn't an `As` token.

Reply:
Also the space might not have been typed yet. The reason for `hasAsToken` is to avoid autocompleting `AsAs Variant`.